guides/common/modules/proc_installing-and-configuring-puppet-agent-during-host-registration.adoc

```diff
@@ -0,0 +1,63 @@
+[id="installing-and-configuring-puppet-agent-during-host-registration_{context}"]
+= Installing and Configuring Puppet Agent during Host Registration
+
+Use this procedure to install and configure the Puppet agent on the host during registration.
+
+ifdef::satellite[]
+.Prerequisites
+* You enabled and synchronized the *{project-client-name}* repository to {Project}.
+For more information, see {ContentManagementDocURL}Importing_Content_content-management[Importing Content] in _{ContentManagementDocTitle}_.
+* You created an activation key that enables the *{project-client-name}* repository for hosts.
+For more information, see {ContentManagementDocURL}Managing_Activation_Keys_content-management[Managing Activation Keys] in _{ContentManagementDocTitle}_.
+endif::[]
+ifdef::katello,orcharhino[]
+.Prerequisites
+* You created a Product and repository for the upstream Puppet agent, such as `\https://yum.puppet.com` or `\https://apt.puppet.com`, and synchronized the repository to {Project}.
+For more information, see {ContentManagementDocURL}Importing_Content_content-management[Importing Content] in _{ContentManagementDocTitle}_.
+* You created an activation key that enables the Puppet agent repository for hosts.
+For more information, see {ContentManagementDocURL}Managing_Activation_Keys_content-management[Managing Activation Keys] in _{ContentManagementDocTitle}_.
+endif::[]
+
+.Procedure
+. In the {ProjectWebUI}, navigate to *Configure* > *Global Parameters* to add host parameters globally.
+Alternatively, you can navigate to *Configure* > *Host Groups* and edit or create a host group to add host parameters only in a host group.
+ifdef::foreman-el,foreman-deb[]
+. Enable the Puppet agent from the official Puppet repository using one of the following host parameters in global parameters or a host group:
+
+* Add a host parameter named `enable-puppetlabs-repo` for the latest stable Puppet release in the unversioned repository.
+* Add a host parameter named `enable-official-puppet7-repo` for the Puppet 7 release repository.
+* Add a host parameter named `enable-puppetlabs-puppet6-repo` for the Puppet 6 release repository.
```

Comment:
By now Puppet 6 is EOL. Given we don't publish guides for `foreman-deb` and `foreman-el` on stable releases I think it'd be safe to leave this line out by now.